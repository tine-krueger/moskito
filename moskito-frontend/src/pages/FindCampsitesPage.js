import PropTypes from 'prop-types'
import { useEffect } from 'react'
import FilterCampsite from '../components/Forms/FilterCampsiteForm'
import Header from '../components/Header/Header'
import Navigation from '../components/Navigation/Navigation'


FindCampsitesPages.propTypes = {
    getCampsites: PropTypes.func.isRequired,
    headline: PropTypes.string.isRequired,
    setLoading: PropTypes.func.isRequired
}

export default function FindCampsitesPages({getCampsites, headline, setLoading}) {
    useEffect(() => {
        setLoading(true)
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])
    
    return (
        <>
            <Header children={headline}/>
            <FilterCampsite getCampsites={getCampsites}/>
            <Navigation/>
        </>
    )   
}
