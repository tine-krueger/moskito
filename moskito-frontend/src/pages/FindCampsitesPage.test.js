import 'jest-styled-components'
import userEvent from '@testing-library/user-event'
import FindCampsitesPage from './FindCampsitesPage'
import renderWithRouter from '../testSetup/setupTests'
import campsiteBasisData from '../data/campsiteBasisData.json'

const mockGetCampsites = jest.fn()
const mockHeadline = 'Headline'
const mockSetLoading = jest.fn()
const mockDeleteTokens = jest.fn()
const mocksetAuthTokens = jest.fn()

jest.mock('../context/auth.js', () => ({
    ...jest.requireActual('../context/auth.js'),
    useAuth: () => ({
        deleteTokens: mockDeleteTokens,
        setAuthTokens: mocksetAuthTokens
    })
}))

const component = <FindCampsitesPage
getCampsites={mockGetCampsites}
headline={mockHeadline}
setLoading={mockSetLoading} 
/>
describe('Find Campsites Page', () => {
    it('renders correctly', () => {
        const { container } = renderWithRouter(component)
        expect(container).toMatchSnapshot()
    })

    it('calls getCampsites when clicking the Button with the right parameter and redirects to results page', () => {
        const { getByText, history } = renderWithRouter(component)
        userEvent.click(getByText('CampingplÃ¤tze finden'))
        expect(mockGetCampsites).toHaveBeenCalledTimes(1)
        expect(mockGetCampsites).toHaveBeenCalledWith(campsiteBasisData)
        expect(history.location.pathname).toBe('/campsites')
    })

    it('navigates to bookmarks-page, when clicking on nav-button', () => {
        const { getByTestId, history } = renderWithRouter(component)
        userEvent.click(getByTestId('bookmarks'))
        expect(history.location.pathname).toBe('/bookmarks')
    })

    it('logs out by deleting tokens, when clicking on nav-button', () => {
        const { getByTestId } = renderWithRouter(component)
        userEvent.click(getByTestId('logout'))
        expect(mocksetAuthTokens).toHaveBeenCalledTimes(1)
        expect(mockDeleteTokens).toHaveBeenCalledTimes(1)
    })
})