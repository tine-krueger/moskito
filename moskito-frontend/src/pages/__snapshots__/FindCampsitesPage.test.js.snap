// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find Campsites Page renders correctly 1`] = `
.c4 {
  position: relative;
  height: 2.5em;
  margin-bottom: .75em;
}

.c4 input[type="checkbox"] {
  opacity: 0;
  position: absolute;
  visibility: hidden;
}

.c4 p {
  margin: 0;
  padding: .1em 0 0 2.5em;
  font-size: 1.1em;
}

.c4 p::before {
  content: " ";
  display: inline-block;
  position: absolute;
  width: 25px;
  height: 25px;
  left: 0;
  top: 0;
  -webkit-transition: all .2s ease;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  border-radius: 5px;
  background-color: var(--background-light);
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c4 p::after {
  content: " ";
  width: 14px;
  height: 14px;
  top: 6px;
  position: absolute;
  opacity: 0;
  background-color: var(--link);
  font-weight: 900;
  border-radius: 2px;
  display: inline-block;
  left: 6px;
}

.c4 input[type="checkbox"]:checked+label>p::after {
  opacity: 1;
}

.c4 input[type="checkbox"]:checked+label>p::before {
  box-shadow: inset 5px 5px 3px -3px rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,.9);
}

.c5 {
  padding: 1em 2em;
  font-family: 'Arvo';
  font-size: 1.1rem;
  border-radius: 10px;
  width: auto;
  border: none;
  background: var(--background);
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  color: var(--headline);
  text-transform: uppercase;
  margin: 1em auto 3em auto;
}

.c5:hover {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  outline: none;
}

.c2 {
  grid-column: 1 / -1;
  width: 100%;
  text-align: center;
  margin: 0 auto 2em;
  margin-bottom: 1em;
}

.c2 h3 {
  text-align: left;
  padding-left: .5em;
  margin: 0 0 1em;
}

.c2 input {
  width: 100%;
  padding: 1em;
  height: 3em;
  font-size: 1em;
  border: none;
  border-radius: 10px;
  color: #6b717e;
  background-color: #d8e6e4;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input:focus {
  outline: none;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3),0 0 0px 1.7px #d3a392;
}

.c2 input[type=range] {
  width: 100%;
  margin: 4.05px 0;
  background-color: transparent;
  -webkit-appearance: none;
  box-shadow: none;
}

.c2 input[type=range]:focus {
  outline: none;
  box-shadow: none;
}

.c2 input[type=range]::-webkit-slider-runnable-track {
  background: var(--background-light);
  border: 0;
  border-radius: 25px;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input[type=range]::-webkit-slider-thumb {
  margin-top: -4.05px;
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  -webkit-appearance: none;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c2 input[type=range]:focus::-webkit-slider-runnable-track {
  background: var(--background-light);
}

.c2 input[type=range]::-moz-range-track {
  background: var(--background-light);
  border: 0;
  border-radius: 25px;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input[type=range]::-moz-range-thumb {
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c2 input[type=range]::-ms-track {
  background: transparent;
  border-color: transparent;
  border-width: 9.65px 0;
  color: transparent;
  width: 100%;
  height: 1.5em;
  cursor: pointer;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c2 input[type=range]::-ms-fill-lower {
  background: var(--background-light);
  border: 0;
  border-radius: 50px;
}

.c2 input[type=range]::-ms-fill-upper {
  background: var(--background-light);
  border: 0;
  border-radius: 50px;
}

.c2 input[type=range]::-ms-thumb {
  width: 2em;
  height: 2em;
  background: var(--link);
  border: 0;
  border-radius: 2em;
  cursor: pointer;
  margin-top: 0px;
  box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
}

.c2 input[type=range]:focus::-ms-fill-lower {
  background: var(--background-light);
}

.c2 input[type=range]:focus::-ms-fill-upper {
  background: var(--background-light);
}

.c1 {
  max-width: 500px;
  display: grid;
  margin: 2em 2em 3em 2em;
}

.c1 h4 {
  padding-left: .5em;
  margin: 0 0 1em;
}

.c3 {
  margin-top: 1em;
  padding-left: .5em;
}

.c0 {
  width: 80%;
  margin: 2em auto 3em;
  padding: .25em .5em;
  border-radius: 20px;
  text-align: center;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
}

.c6 {
  margin: 0 auto;
  max-width: 500px;
  background-color: var(--background);
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
}

.c7 {
  padding-left: 0;
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c8 {
  text-align: center;
  border-radius: 10px;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-1px -1px 2px 0px rgba(255,255,255,0.5);
  width: 2.3em;
  height: 2.3em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 button {
  background-color: transparent;
  border: none;
}

.c8 button:hover {
  outline: none;
}

.c8 a svg,
.c8 button svg {
  width: 1.5em;
  height: 1.5em;
  stroke: var(--link);
}

.c8 button svg {
  width: 1.7em;
  height: 1.7em;
}

.c8:hover,
.c8:active,
.c8:focus {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  outline: none;
}

@supports (-ms-ime-align:auto) {
  .c2 input[type=range] {
    margin: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <h1>
      Headline
    </h1>
  </div>
  <form
    class="c1"
  >
    <label
      class="c2"
      data-testid="input-field"
    >
      <h3>
        Dein Ziel:
      </h3>
      <input
        autocomplete="off"
        data-testid="postalCode"
        name="postalCode"
        placeholder="PLZ/Ort"
        type="text"
        value=""
      />
    </label>
    <label
      class="c2"
      data-testid="input-field"
    >
      <h3>
        Umkreis in km: 
        100
      </h3>
      <input
        data-testid="distance"
        max="200"
        min="5"
        name="distance"
        type="range"
        value="100"
      />
    </label>
    <h3>
      (Keine) Ausstattung:
    </h3>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <input
          data-testid="wlan"
          id="wlan"
          name="wlan"
          type="checkbox"
        />
        <label
          for="wlan"
        >
          <p>
            kein W-Lan
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="music"
          id="music"
          name="music"
          type="checkbox"
        />
        <label
          for="music"
        >
          <p>
            keine Bar/Diskothek
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="animation"
          id="animation"
          name="animation"
          type="checkbox"
        />
        <label
          for="animation"
        >
          <p>
            keine Animation
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="fire"
          id="fire"
          name="fire"
          type="checkbox"
        />
        <label
          for="fire"
        >
          <p>
            Lagerfeuerstelle
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="path"
          id="path"
          name="path"
          type="checkbox"
        />
        <label
          for="path"
        >
          <p>
            Kies- oder Sandwege
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="bulli"
          id="bulli"
          name="bulli"
          type="checkbox"
        />
        <label
          for="bulli"
        >
          <p>
            Bulli-Wiese
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="tents"
          id="tents"
          name="tents"
          type="checkbox"
        />
        <label
          for="tents"
        >
          <p>
            Zeltbereiche
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="subdevision"
          id="subdevision"
          name="subdevision"
          type="checkbox"
        />
        <label
          for="subdevision"
        >
          <p>
            keine Parzellierung
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="permanent"
          id="permanent"
          name="permanent"
          type="checkbox"
        />
        <label
          for="permanent"
        >
          <p>
            wenig/keine Dauercamper
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="size"
          id="size"
          name="size"
          type="checkbox"
        />
        <label
          for="size"
        >
          <p>
            mind. 80qm Standfläche
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="bio"
          id="bio"
          name="bio"
          type="checkbox"
        />
        <label
          for="bio"
        >
          <p>
            Bio/Regionale Lebensmittel
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="snack"
          id="snack"
          name="snack"
          type="checkbox"
        />
        <label
          for="snack"
        >
          <p>
            Imbiss
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="animals"
          id="animals"
          name="animals"
          type="checkbox"
        />
        <label
          for="animals"
        >
          <p>
            Hunde erlaubt
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="seaside"
          id="seaside"
          name="seaside"
          type="checkbox"
        />
        <label
          for="seaside"
        >
          <p>
            Zugang zum See/Meer
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="bathing"
          id="bathing"
          name="bathing"
          type="checkbox"
        />
        <label
          for="bathing"
        >
          <p>
            Badestelle für Kinder
          </p>
        </label>
      </div>
      <div
        class="c4"
      >
        <input
          data-testid="forest"
          id="forest"
          name="forest"
          type="checkbox"
        />
        <label
          for="forest"
        >
          <p>
            mit Wald
          </p>
        </label>
      </div>
    </div>
    <button
      class="c5"
    >
      Campingplätze finden
    </button>
  </form>
  <nav
    class="c6"
  >
    <ul
      class="c7"
    >
      <li
        class="c8"
      >
        <a
          href="/find-campsite"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </a>
      </li>
      <li
        class="c8"
      >
        <a
          data-testid="bookmarks"
          href="/bookmarks"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M878.3 392.1L631.9 145.7c-6.5-6.5-15-9.7-23.5-9.7s-17 3.2-23.5 9.7L423.8 306.9c-12.2-1.4-24.5-2-36.8-2-73.2 0-146.4 24.1-206.5 72.3-15.4 12.3-16.6 35.4-2.7 49.4l181.7 181.7-215.4 215.2a15.8 15.8 0 0 0-4.6 9.8l-3.4 37.2c-.9 9.4 6.6 17.4 15.9 17.4.5 0 1 0 1.5-.1l37.2-3.4c3.7-.3 7.2-2 9.8-4.6l215.4-215.4 181.7 181.7c6.5 6.5 15 9.7 23.5 9.7 9.7 0 19.3-4.2 25.9-12.4 56.3-70.3 79.7-158.3 70.2-243.4l161.1-161.1c12.9-12.8 12.9-33.8 0-46.8z"
            />
          </svg>
        </a>
      </li>
      <li
        class="c8"
      >
        <button
          data-testid="logout"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
